ast/src/test/java/com/graphicsfuzz/common/typing/TyperTest.java

```diff
@@ -1000,6 +1000,56 @@ public void visitBinaryExpr(BinaryExpr binaryExpr) {
     }.visit(tu);
   }
 
+  @Test
+  public void testScalarSwizzleTyped() throws Exception {
+    final TranslationUnit tu = ParseHelper.parse("#version 420\n"
+        + "void main() {\n"
+        + "  float f;\n"
+        + "  vec3 fv;\n"
```

Comment:
Should we also test vec4, and all swizzle letters `xyzw`, `rgba` and `stpq` ?

Reply:
Similar.